components/org.wso2.carbon.crypto.provider.hsm/src/main/java/org/wso2/carbon/crypto/provider/hsm/PKCS11JCEObjectMapper.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.crypto.provider.hsm;
+
+import iaik.pkcs.pkcs11.objects.Certificate;
+import iaik.pkcs.pkcs11.objects.PrivateKey;
+import iaik.pkcs.pkcs11.objects.RSAPrivateKey;
+import iaik.pkcs.pkcs11.objects.RSAPublicKey;
+import iaik.pkcs.pkcs11.objects.X509PublicKeyCertificate;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.crypto.api.CryptoException;
+
+import java.io.ByteArrayInputStream;
+import java.math.BigInteger;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.RSAPrivateKeySpec;
+
+/**
+ * This class maps PKCS 11 certificates and private keys to JCE certificates and private keys and vice versa.
+ */
+public class PKCS11JCEObjectMapper {
+
+    private static Log log = LogFactory.getLog(PKCS11JCEObjectMapper.class);
+
+    /**
+     * This static method maps JCE certificate to PKCS 11 certificate.
+     *
+     * @param certificate : JCE Certificate
+     * @return PKCS11 Certificate data {@link PKCS11CertificateData}.
+     * @throws CryptoException
+     */
+    public static PKCS11CertificateData mapCertificateJCEToPKCS11(java.security.cert.Certificate certificate)
+            throws CryptoException {
+
+        if (!(certificate instanceof X509Certificate)) {
+            String errorMessage = "This doesn't support for conversion of certificates in the provided format.";
```

Comment:
May be we can say what the provided format is. Let's try always to be descriptive. Give as much as information you can, so that the person who is running this code will infer more about the issues taking place inside.